fml/synchronization/sync_switch.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_SYNCHRONIZATION_SYNC_SWITCH_H_
+#define FLUTTER_FML_SYNCHRONIZATION_SYNC_SWITCH_H_
+
+#include <forward_list>
+#include <functional>
+#include <mutex>
+
+#include "flutter/fml/macros.h"
+
+namespace fml {
+
+/// A threadsafe structure that allows you to switch between 2 different
+/// execution paths.
+///
+/// Execution and setting the switch is exclusive, i.e. only one will happen
+/// at a time.
+class SyncSwitch {
+ public:
+  /// Represents the 2 code paths available when calling |SyncSwitch::Execute|.
+  struct Handlers {
+    /// Sets the handler that will be executed if the |SyncSwitch| is true.
+    Handlers& SetIfTrue(std::function<void()>&& handler);
```

Comment:
Still neither here not there, either use the pass by value if using the convention before https://github.com/flutter/engine/pull/13975 lands or const refs if after.

Reply:
Discussed offline.  I was working under the false idea that you had to explicitly create rvalues and that this would force users to pass in rvalues.  I moved this over to const ref to match the other code.